ts/sagas/wallet.ts

```diff
@@ -861,19 +882,21 @@ function* continueWithPaymentMethodsHandler(
   if (!hasPaymentId && sessionToken !== undefined) {
     try {
       yield put(paymentSetLoadingState());
-      const tmp: string | undefined = yield call(
+      const result: Either<NodoErrors, string> = yield call(
         attivaAndGetPaymentId,
         postAttivaRpt,
         getPaymentIdApi,
         rptId,
         paymentContextCode,
         amount
       );
-      paymentId = tmp;
+      if (result.isRight()) {
+        paymentId = result.value;
+      } else {
+        yield put(paymentFailure(result.value));
+      }
     } catch {
-      /**
-       * TODO handle error
-       */
+      yield put(paymentFailure("GENERIC_ERROR"));
     } finally {
       yield put(paymentResetLoadingState());
```

Comment:
the order of the actions looks off: `paymentFaulure` is triggered before `resetLoadingState` (shouldn't be the other way around? first loading stops, then validation occurs, then success/failure)

Reply:
the problem is not that `put`s are blocking, but that they are processed in the order of dispatch by the reducers, so state gets updated in the order of dispatch and in turn components get updated in the order of dispatch - it's ok for now, but it's nevertheless incorrect, so we will fix it in the future